lib/pi_piper/sysfs/driver.rb

```diff
@@ -0,0 +1,106 @@
+require 'filewatcher'
+
+module PiPiper
+  module Sysfs
+    class Driver < PiPiper::Driver
+
+      GPIO_HIGH = 1
+      GPIO_LOW  = 0
+
+      def initialize
+        @exported_pins = Set.new
+      end
+
+      def close
+        unexport_all
+        @exported_pins.empty? 
+      end
+      
+  # Support GPIO pins
+      def pin_direction(pin, direction)
+        raise ArgumentError, "direction should be :in or :out" unless [:in, :out].include? direction
+        export(pin)
+        raise RuntimeError, "Pin #{pin} not exported" unless exported?(pin)
+        File.write(direction_file(pin), direction)
+      end
+
+      def pin_read(pin)
+        raise ArgumentError, "Pin #{pin} not exported" unless exported?(pin)
+        File.read(value_file(pin)).to_i
+      end
+
+      def pin_write(pin, value)
+        raise ArgumentError, "value should be GPIO_HIGH or GPIO_LOW" unless [GPIO_LOW, GPIO_HIGH].include? value
+        raise ArgumentError, "Pin #{pin} not exported" unless exported?(pin)
+        File.write(value_file(pin), value)
+      end
+
+      def pin_set_pud(pin, value)
+        raise NotImplementedError, "Pull up/down not available with this driver. keep it on :off" unless value == :off
+      end
+      
+      def pin_set_trigger(pin, trigger)
+        raise ArgumentError, "trigger should be :falling, :rising, :both or :none" unless [:falling, :rising, :both, :none].include? trigger
+        raise ArgumentError, "Pin #{pin} not exported" unless exported?(pin)
+        File.write(edge_file(pin), trigger)
+      end
+
+      def pin_wait_for(pin, trigger)
+        pin_set_trigger(pin, trigger)
+        value = pin_read(pin)
+
```

Comment:
I was not aware of any CPU usage issue with the previous implementation, it was backed by a unix system call `select()` which is widely used since august 1984... and worked pretty well since I used PiPiper also (maybe a year).
can you please explain me a bit more the issue ?

on your implementation : This need to be discussed ; your are watching the value_file and not the edge_file, why ? Is it wise to add a new dependecy to the gem ? FileWatcher seems to use File#mtime to check if a file is changed, is it reliable ? nothing in the gpio kernel module is said about dates of the value or edge files.

Let's talk.